website/client/src/components/notifications.vue

```diff
@@ -893,55 +745,69 @@ export default {
-          case 'ACHIEVEMENT_WILD_BLUE_YONDER':
-          case 'ACHIEVEMENT_DOMESTICATED':
-          case 'ACHIEVEMENT_SHADY_CUSTOMER':
-          case 'ACHIEVEMENT_SHADE_OF_IT_ALL':
+          case 'ACHIEVEMENT_PARTY_UP':
+          case 'ACHIEVEMENT_ULTIMATE_GEAR':
           case 'GENERIC_ACHIEVEMENT':
             this.showNotificationWithModal(notification);
             break;
+          case 'ACHIEVEMENT_QUESTS': {
+            const { achievement } = notification.data;
+            const upperCaseAchievement = achievement.charAt(0).toUpperCase() + achievement.slice(1);
+            const achievementTitleKey = `achievement${upperCaseAchievement}`;
+            NOTIFICATIONS.ACHIEVEMENT_QUESTS.label = $t => `${$t('achievement')}: ${$t(achievementTitleKey)}`;
+            this.showNotificationWithModal(notification);
+            Vue.set(this.user.achievements, achievement, true);
+            break;
+          }
+          case 'ACHIEVEMENT_STABLE': {
+            const { achievement, achievementNotification } = notification.data;
+            NOTIFICATIONS.ACHIEVEMENT_STABLE.label = $t => `${$t('achievement')}: ${$t(achievementNotification)}`;
+            this.showNotificationWithModal(notification);
+            Vue.set(this.user.achievements, achievement, true);
+            break;
+          }
+          case 'ACHIEVEMENT_ANIMAL_SET': {
+            const { achievement } = notification.data;
+            const upperCaseAchievement = achievement.charAt(0).toUpperCase() + achievement.slice(1);
+            const achievementTitleKey = `achievement${upperCaseAchievement}`;
+            NOTIFICATIONS.ACHIEVEMENT_ANIMAL_SET.label = $t => `${$t('achievement')}: ${$t(achievementTitleKey)}`;
+            this.showNotificationWithModal(notification);
+            Vue.set(this.user.achievements, achievement, true);
+            break;
+          }
+          case 'ACHIEVEMENT_PET_COLOR': {
+            const { achievement } = notification.data;
+            const upperCaseAchievement = achievement.charAt(0).toUpperCase() + achievement.slice(1);
+            const achievementTitleKey = `achievement${upperCaseAchievement}`;
+            NOTIFICATIONS.ACHIEVEMENT_PET_COLOR.label = $t => `${$t('achievement')}: ${$t(achievementTitleKey)}`;
+            this.showNotificationWithModal(notification);
+            Vue.set(this.user.achievements, achievement, true);
+            break;
+          }
+          case 'ACHIEVEMENT_MOUNT_COLOR': {
+            const { achievement } = notification.data;
+            const upperCaseAchievement = achievement.charAt(0).toUpperCase() + achievement.slice(1);
+            const achievementTitleKey = `achievement${upperCaseAchievement}`;
+            NOTIFICATIONS.ACHIEVEMENT_MOUNT_COLOR.label = $t => `${$t('achievement')}: ${$t(achievementTitleKey)}`;
+            this.showNotificationWithModal(notification);
+            Vue.set(this.user.achievements, achievement, true);
+            break;
+          }
```

Comment:
continuing from above, see how these are all still almost identical? hmm...

I may poke at this if you're not feeling up to it!

Reply:
Yeah, I see what you're getting at--and I can see how we could collapse these down into one--I'll give this one a try and see how it goes. I'm not quite seeing how it would work in the code above where we're defining the different achievements as they're pulling in data from different constant files `stableAchievs`, `questSeriesAchievs`, `animalSetAchievs` etc--part of the goal here was to make it easier to maintain each type of achievement instead of having one big list.

Reply:
Hmmm. Now that I look at this more closely, I don't think I can collapse this, as it's pulling the achievement strings from each of the constant files... Let's discuss this.